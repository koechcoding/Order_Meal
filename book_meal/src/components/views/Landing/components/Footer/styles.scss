
footer {
    margin-top: 40px;
    min-height: 250px;
}
footer p {
    color: white;
    width: 100%;
    margin: auto;
    background-color: rgba(0, 80, 80, 0.4); 
  }