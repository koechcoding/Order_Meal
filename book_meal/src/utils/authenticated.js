export default function authenticated(){
    try{
        return JSON.parse(localStorage.getItem('user'));
    } catch(err){
        
    }
}