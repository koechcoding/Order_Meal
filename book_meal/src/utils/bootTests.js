import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter()});

global.window = {}
import localStorage from 'mock-local-Storage';
window.localStorage = global.localStorage

export default Enzyme;
