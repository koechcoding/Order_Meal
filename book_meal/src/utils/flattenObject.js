export default function flattenObject(obj){
    let result = {}
    for (const key in obj){
        if((typeof obj[key])=== 'object' &&  obj[key]  !== null) {
            let flattened = flattenObject(obj[key])
            for(const key2 in flattened){
                result[key + "." + key2] = flattened[key2];
            }
        }else{
            result[key] = obj[key];
        }
    }
    return result;
}