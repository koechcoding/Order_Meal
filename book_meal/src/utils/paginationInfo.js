export default function paginationInfo(page){

    if(!page){
        return{}
    }

    let currentPage = 1
    if(page.next_page)
         currentPage = page.next_page -1;
    else if (page.prev_page)
        currentPage = page.prev_page + 1;

    let pageCount = Math.ceil(page.total / page.per_paage);
    if(Number.isNaN(pageCount) || pageCount === 0)
      pageCount = 1; 

    return {
        pageCount,
        currentPage,
        total: page.total,
        perPage: page.per_page,
        hasNext: page.has_next,
        hasPrev: page.has_prev,
        nextPage: page.next_page,
        prevPage: page.prev_page,
        currentCount: page.current_count,
    };
}