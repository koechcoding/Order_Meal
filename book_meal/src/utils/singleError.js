const singleError = (errorPayload)=>{
    if(errorPayload && errorPayload.data){
        const response = errorPayload.data;
        //alternative to Object.keys...
        for(let errorKey in response.errors){
            return response.error[errorKey]
        }
        if(response.message){
            return response.message;
        }
    }
    return 'Unknown error. Please try again!';
}
export default singleError;