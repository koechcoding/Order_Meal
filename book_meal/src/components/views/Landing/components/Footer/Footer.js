import React from 'react';
import './styles.css';

class Footer extends React.Component {
    render() {
        return (
            <footer className="container-fluid bg-teal patterns mt-5 text-center p-5">
                <p className="p-4 m-auto text-center w-50">
                    Andela Developer Challenge<br/>
                    Made with &lt;3 by Kelvin Koech.
                </p>
            </footer>
        );
    }
}

export default Footer