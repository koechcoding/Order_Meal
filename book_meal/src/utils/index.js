import singleError from './singleError';
import paginationInfo from './paginationInfo';
import authenticated from './authenticated';
import flattenObject from './flattenObject';

export { 
    singleError, 
    paginationInfo, 
    authenticated,
    flattenObject
}