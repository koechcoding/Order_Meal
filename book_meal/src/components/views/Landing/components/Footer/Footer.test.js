import React from 'react';
import { shallow }  from 'enzyme';
import 'src/utils/bootTests'; 
import Footer from './Footer';

describe('<Footer />', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<Footer/>);
        expect(wrapper).toMatchSnapshot();
    });
});