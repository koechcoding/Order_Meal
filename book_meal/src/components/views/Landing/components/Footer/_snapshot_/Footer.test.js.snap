// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p
        className="p-4 m-auto text-center w-50"
>
        Andela Developer Challenge
        <br />
        Made with &lt;3 by Edward Njoroge.
</p>,
      "className": "container-fluid bg-teal patterns mt-5 text-center p-5",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Andela Developer Challenge",
          <br />,
          "Made with <3 by Edward Njoroge.",
        ],
        "className": "p-4 m-auto text-center w-50",
      },
      "ref": null,
      "rendered": Array [
        "Andela Developer Challenge",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        "Made with <3 by Edward Njoroge.",
      ],
      "type": "p",
    },
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p
          className="p-4 m-auto text-center w-50"
>
          Andela Developer Challenge
          <br />
          Made with &lt;3 by Edward Njoroge.
</p>,
        "className": "container-fluid bg-teal patterns mt-5 text-center p-5",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Andela Developer Challenge",
            <br />,
            "Made with <3 by Edward Njoroge.",
          ],
          "className": "p-4 m-auto text-center w-50",
        },
        "ref": null,
        "rendered": Array [
          "Andela Developer Challenge",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "Made with <3 by Edward Njoroge.",
        ],
        "type": "p",
      },
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;